File: 3876. Construct Uniform Parity Array II/Solution.py
class Solution(object):
    def uniformArray(self, nums1):
        """
        :type nums1: List[int]
        :rtype: bool
        """

        minOdd = float('inf')
        nums2 = [0]*len(nums1)

        for i in range(len(nums1)):
            if(nums1[i]%2 != 0):
                minOdd = min(nums1[i],minOdd)

        
        if(minOdd == float('inf')):
            return True

        for i in range(len(nums1)):
            if(nums1[i]%2==0):
                nums2[i] = nums1[i] - minOdd

        for num in nums2:
            if(num<0):
                return False

        return True


        

File: 3876. Construct Uniform Parity Array II/Solution.java
//best Solution

class Solution {
    public boolean uniformArray(int[] nums1) {

        //find min Value 
        int min = Integer.MAX_VALUE;
        for(int num:nums1){
            min = Math.min(min,num);
        }

        //if min value is odd the ans is always True
        if (min%2!=0){
            return true;
        }


        //if min value is even we need to check all the elements should be even 
        for(int num : nums1){
            if(num%2!=0){
                return false;
            }
        }
        
        return true;
    }
}